src/commands/transaction/get.js

```diff
@@ -43,40 +50,110 @@ const queryNode = async (client, txnState, parameters) =>
 
 export default class GetCommand extends BaseCommand {
 	async run() {
-		const { args: { ids }, flags: { state: txnState } } = this.parse(
-			GetCommand,
-		);
-		const req = ids.map(id => ({
-			query: {
-				limit: 1,
-				id,
-			},
-			placeholder: {
-				id,
-				message: 'Transaction not found.',
-			},
-		}));
+		const {
+			args: { ids },
+			flags: { state: txnState, limit, offset, senderId: senderAddress },
+		} = this.parse(GetCommand);
 
 		const client = getAPIClient(this.userConfig.api);
 
-		if (txnState === 'unsigned') {
-			const results = await queryNode(client.node, txnState, req);
+		if (txnState) {
+			if (ids) {
+				const reqTransactionIDs = ids.map(id => ({
+					query: {
+						limit: 1,
+						id,
+					},
+					placeholder: {
+						id,
+						message: 'Transaction not found.',
+					},
+				}));
+
+				const results = await queryNode(
+					client.node,
+					txnState,
+					reqTransactionIDs,
+				);
+				return this.print(results);
+			}
+
+			if (senderAddress) {
+				const reqWithSenderId = [
+					{
+						query: {
+							limit,
+							offset,
+							senderId: senderAddress,
+						},
+						placeholder: {
+							senderId: senderAddress,
+							message: 'Transaction not found.',
+						},
+					},
+				];
+
+				const results = await queryNode(client.node, txnState, reqWithSenderId);
+				return this.print(results);
+			}
+
+			const reqByLimitOffset = [
+				{
+					query: {
+						limit,
+						offset,
+					},
+					placeholder: {
+						message: 'No transactions found.',
+					},
+				},
+			];
+
+			const results = await queryNode(client.node, txnState, reqByLimitOffset);
+
 			return this.print(results);
 		}
-		if (txnState === 'unprocessed') {
-			const results = await queryNode(client.node, txnState, req);
+
+		if (ids) {
+			const reqTransactionIDs = ids.map(id => ({
+				query: {
+					limit: 1,
+					id,
+				},
+				placeholder: {
+					id,
+					message: 'Transaction not found.',
+				},
+			}));
+			const results = await query(client, 'transactions', reqTransactionIDs);
+
 			return this.print(results);
 		}
+
+		const req = {
+			query: {
+				limit,
+				offset,
+			},
+			placeholder: {
+				message: 'No transactions found.',
+			},
+		};
 		const results = await query(client, 'transactions', req);
 
-		return this.print(results);
+		if (!isArray(results)) {
+			return this.print(results);
+		}
+		const sortedResults = results.sort((a, b) => b.height - a.height);
```

Comment:
In stead of doing `sort` in the code, probably better to use API sort field

Reply:
@shuse2 I just checked the API and actually, it doesn't support sort by height. `Available values : amount:asc, amount:desc, fee:asc, fee:desc, type:asc, type:desc, timestamp:asc, timestamp:desc. Default value : amount:asc`
We can also keep it default that will sort it by amount or we keep it by height?